src/Package/Impl/Shell/VsServiceManager.cs

```diff
@@ -2,33 +2,53 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+using System.ComponentModel.Composition;
+using System.ComponentModel.Composition.Hosting;
 using Microsoft.Common.Core.Services;
 using Microsoft.Common.Core.Shell;
+using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.R.Packages.R;
 using VsPackage = Microsoft.VisualStudio.Shell.Package;
 
 namespace Microsoft.VisualStudio.R.Package.Shell {
     internal sealed class VsServiceManager : ServiceManager {
-        private readonly IApplicationShell _appShell;
+        private readonly ICoreShell _shell;
+        private readonly ICompositionService _compositionService;
+        private readonly ExportProvider _exportProvider;
 
-        public VsServiceManager(IApplicationShell appShell) {
-            _appShell = appShell;
+        public VsServiceManager(ICoreShell shell) {
+            _shell = shell;
+            var componentModel = (IComponentModel)VsPackage.GetGlobalService(typeof(SComponentModel));
+            _compositionService = componentModel.DefaultCompositionService;
+            _exportProvider = componentModel.DefaultExportProvider;
         }
 
         #region IServiceContainer
         public override T GetService<T>(Type type = null) {
+            type = type ?? typeof(T);
+
+            if(type == typeof(ExportProvider)) {
```

Comment:
Why do you want to make it a special case instead of using `AddService` for these three services?